Partner Docs: Incentive management guides

Xeonus opened the pull request:
- Showcase how incentive management works on Balancer
- Link relevant operations UI components
- Can potentially add more refacorings / reorgs to this PR as some sections in v2 are now general

Xeonus committed:
feat: add Incentive mgmt page

- checkpoint for incentive mgmt page contents

Xeonus committed:
feat: add sections

Xeonus committed:
feat: add injector config docs

Xeonus committed:
feat: add checklist functionality

- add secondary reward token checklist

Xeonus committed:
fix: remove redundant content

mkflow27 commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
```diff
@@ -0,0 +1,123 @@
+BAL is emitted to whitelisted staking gauges in our veBAL system. For a pool to be eligible for BAL rewards, it needs to be voted in by governance. The process on how to apply for a BAL staking gauge can be found [here](../balancer-v2/gauge-onboarding.md). A guide on how to set up a gauge can be found in the gauge creation [section](incentive-management.md#gauge-creation)
+
+## Secondary Reward Token Incentives
+The Balancer Maxis have built a sophisticated infrastructure to create and manage secondary reward campaigns for Balancer staking gauges. To make full use of this system, the Maxis provide tooling to facilitate the setup.
```
Could potentially add a separate callout that tells me that all? below steps can be managed on the operations UI (small explainer of why it is named that way?) and what the "operations UI" is. While I know that operations UI and balancer.defilytica.tools is the same (due to the presentation last week) it might not be that obvious to the reader.

mkflow27 commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
```diff
@@ -0,0 +1,123 @@
+BAL is emitted to whitelisted staking gauges in our veBAL system. For a pool to be eligible for BAL rewards, it needs to be voted in by governance. The process on how to apply for a BAL staking gauge can be found [here](../balancer-v2/gauge-onboarding.md). A guide on how to set up a gauge can be found in the gauge creation [section](incentive-management.md#gauge-creation)
+
+## Secondary Reward Token Incentives
+The Balancer Maxis have built a sophisticated infrastructure to create and manage secondary reward campaigns for Balancer staking gauges. To make full use of this system, the Maxis provide tooling to facilitate the setup.
```
suggestion: Could potentially add a separate callout that tells me that all? below steps can be managed on the operations UI (small explainer of why it is named that way?) and what the "operations UI" is. While I know that operations UI and balancer.defilytica.tools is the same (due to the presentation last week) it might not be that obvious to the reader.

Xeonus replied:
good point, making it more clear

mkflow27 commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
suggestion: Maybe link to the tools entry or add a link as well?

Xeonus replied:
done, refactored the checklist component to allow for internal and external links

mkflow27 reviewed (approved):
Lgtm. Feel free to merge at your discretion

Xeonus committed:
feat: implement mkflow feedback

- add banner for operations ui
- modify checklist component to allow for internal and external links

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
```diff
@@ -0,0 +1,126 @@
+The Balancer ecosystem is utilizing a modified version of Curve's vyper gauge infrastructure. Here, we outline how you can utilize our staking gauge system. We cover both how you can apply for a veBAL gauge to receive BAL rewards and how secondary reward programs can be set up.
+
+## BAL Incentives through veBAL Gauges
+BAL is emitted to whitelisted staking gauges in our veBAL system. For a pool to be eligible for BAL rewards, it needs to be voted in by governance. The process on how to apply for a BAL staking gauge can be found [here](../balancer-v2/gauge-onboarding.md). A guide on how to set up a gauge can be found in the gauge creation [section](incentive-management.md#gauge-creation)
```
whitelisted? you mean gauges added to/enabled on the gauge controller?

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
```diff
@@ -0,0 +1,126 @@
+::: info
+The Balancer Maxis are at your service to setup and deploy rewards injectors. You can also manage injectors yourself if you please to do so, and we are happy to help along the setup process
+:::
+### Token Whitelisting
```
suggesting to number all the steps; `### Step 1: Token Whitelisting`

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
```diff
@@ -0,0 +1,126 @@
+3. If a gauge has already been created, you can skip this step (UI will show a warning)
+   
+4. If no gauge entry was found for your pool, select it
+5. Depending on your pool composition, select a voting cap above 2% (please refer to gauge caps [here](../balancer-v2/gauge-onboarding.md))
```
this is unclear; the ui recommends 2% cap but here you say to select a voting cap >2%

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
```diff
@@ -0,0 +1,126 @@
+5. There is no need to create a root gauge - only do this if you plan on applying for a veBAL gauge to receive BAL rewards! More detailed instructions on this process can be found [here](https://forum.balancer.fi/t/instructions-overview/2674).
+
+### Rewards Injector Creation
+Depending on your use-case you want to create a rewards injector for your reward token. In that case, you need to follow a series of configuration steps outlined below.
```
this could use more explaination; why would i or would i not want to deploy an injector? what does it do?

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
redundant? this was already done during step 1

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
i would make this the next step after gauge creation. steps 4/5/6 are then optional and strictly related to the injector

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
```suggestion
A gauge can only receive secondary token rewards from a registered `distributor`. On the gauge contract you can read the current configuration via the `reward_data` field by passing the reward token address as input argument. If your reward token is not registered, follow these steps:
```
- pool does not receive rewards; gauge does
- id refrain from also using 'whitelisting' language here; it is confusing wrt the tokenlist whitelisting

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
oh this is great!! maybe move it to the top? link the steps to the respective sub headers?

Xeonus replied:
Keeping at this location as it is crucial this is done last. Added a note / warning of why

gosuto-inzasheru commented on docs/partner-onboarding/onboarding-overview/incentive-management.md:
use same numbered steps here too for clarity/reference

Xeonus committed:
feat: improve incentive section

- resolve feedback
- reorder gauge faq as this is protocol version agnostic
- add numbers to steps so it is easy to follow and reference
- improve checklist styling
